src/rank_llm/demo/rerank_inline_hits.py

```diff
@@ -70,14 +68,6 @@
     },
 ]
 
-results = retrieve_and_rerank(
-    model_path="castorini/rank_zephyr_7b_v1_full",
-    dataset=hits,
-    retrieval_mode=RetrievalMode.QUERY_AND_HITS,
-    retrieval_method=RetrievalMethod.UNSPECIFIED,
-    top_k_candidates=len(hits),
-    print_prompts_responses=True,
-    query=query,
-    variable_passages=True,
-    system_message="You are RankLLM, an intelligent assistant that can rank passages based on their relevancy to the query.",
-)
+retrieved_results = Retriever.from_inline_hits(query=query, hits=hits)

```

Comment:
ah, the difference between "inline hits" and "inline docs" is that the first has docid, etc.

hrm... I'm wondering how to reconcile... maybe inline docs is just inline hits with missing fields (e.g., missing docid?)

Reply:
yes, docs don't have many information that hits do: 1- docid, 2- quid 3- rank 4- score.
In the retriever code in this case, I populate them with best effort mode: doc_id is just the value of the for loop counter (i.e. the order of the document in the document list). 2- quid is one since we have a single query 3- rank similar to docid
4- score left blank